logpuzzle.py

```diff
@@ -40,14 +75,25 @@ def download_images(img_urls, dest_dir):
     with an img tag to show each local image file.
     Creates the directory if necessary.
     """
-    # +++your code here+++
-    pass
+
+    os.mkdir(dest_dir)
+
+    with open('index.html', 'w') as index:
+        img_temp_list = []
+        for i, url in enumerate(img_urls):
+            urllib.urlretrieve(url, 'img{}'.format(i))
+            shutil.move('img{}'.format(i), '{}'.format(dest_dir))
```

Comment:
urlib.urlretrieve streams files to the current working directory. Instead of writing each image to the curent directory then moving it to the destionation directory, save some disk writes by using chdir of the os module to change the working directory of the current process to the destination directory before downloading.
https://pymotw.com/3/os/#managing-the-process-working-directory